phoenix-core/src/it/java/org/apache/phoenix/end2end/BlobTypeIT.java

```diff
@@ -38,19 +41,30 @@ public static synchronized void doSetup() throws Exception {
     }
 
     @Test
-    public void testBlobUpsertFromImageFile() throws SQLException, FileNotFoundException {
+    public void testBlobUpsertFromImageFile() throws SQLException {
         String fName = "/images/dali.jpeg";
         InputStream input = BlobTypeIT.class.getResourceAsStream(fName);
         String tableName = generateUniqueName();
-        Connection conn = DriverManager.getConnection(getUrl());
-        Statement stmt = conn.createStatement();
-        stmt.execute("CREATE TABLE " + tableName + " (id VARCHAR(255) not null primary key, "
-                + "image BLOB)");
-        String upsertQuery = "UPSERT INTO " + tableName + " VALUES(?, ?)";
-        PreparedStatement prepStmt = conn.prepareStatement(upsertQuery);
-        prepStmt.setString(1, "ID1");
-        prepStmt.setBlob(2, input);
-        prepStmt.execute();
+        try (Connection conn = DriverManager.getConnection(getUrl())) {
+            try (Statement stmt = conn.createStatement()) {
+                stmt.execute("CREATE TABLE " + tableName + " (id VARCHAR(255) not null primary key, "
+                        + "image BLOB)");
+            }
+            String upsertQuery = "UPSERT INTO " + tableName + " VALUES(?, ?)";
+            try (PreparedStatement prepStmt = conn.prepareStatement(upsertQuery)) {
+                prepStmt.setString(1, "ID1");
+                prepStmt.setBlob(2, input);
+                prepStmt.execute();
+            }
+            conn.commit();
+        }
+        try (Connection conn = DriverManager.getConnection(getUrl());
+                Statement stmt = conn.createStatement()) {
+            ResultSet rs = stmt.executeQuery("SELECT image FROM " + tableName);
+            assertTrue(rs.next());
+            InputStream inputStr = rs.getBinaryStream(1);
+            assertNotNull(inputStr);
```

Comment:
I'll modify this to do rs.getBlob and check the PheonixBlob once https://github.com/dbwong/phoenix/pull/5 is merged.